packages/shared/lib/core/wallet/utils/getDepositAddress.ts

```diff
@@ -1,14 +1,21 @@
 import { AccountAddress, AccountOutput } from '@iota/sdk/out/types'
 import { IWallet } from '../../profile/interfaces'
 import { getBech32AddressFromAddressTypes } from './getBech32AddressFromAddressTypes'
+import { get } from 'svelte/store'
+import { selectedWalletMainAccountId, updateSelectedWalletMainAccountId } from '../stores'
+import { getBlockIssuerAccounts } from './outputs'
 
-// TODO(2.0): Update the implementation to handle multiple accounts and use the selected account's address.
 export async function getDepositAddress(wallet: IWallet): Promise<string> {
-    const accounts = await wallet?.accounts()
-    if (accounts?.length) {
-        const accountId = (accounts[0]?.output as AccountOutput).accountId
+    const accountOutputsWithBlockIssuerFeature = await getBlockIssuerAccounts(wallet)
+    const _selectedWalletMainAccountId = get(selectedWalletMainAccountId)
```

Comment:
the deposit address should not dependend on the selected wallet, should be agnostic, as it was before